pmd-java/src/main/resources/rulesets/java/controversial.xml

```diff
@@ -379,7 +379,7 @@ adverse impacts on performance.
             <property name="xpath">
                 <value>
                     <![CDATA[
-            //PrimitiveType[@Image = 'short'][name(../..) != 'CastExpression']
+            //PrimitiveType[@Image = 'short'][name(..) != 'CastExpression']
```

Comment:
the downside is the change on the generated AST can impact rules such as this one... I'm therefore not sure to which versions of PMD we may want to merge this.... or if we want to merge it at all, or just keep looking for a way to get `CastExpression` to be parsed properly.